### PR TITLE
[components architecture and urdf examples] Add an example of a 12 DO…

### PR DESCRIPTION
**Pull request by olivier-stasse**
…Fs legged robot

This example is based on a real life equivalent of Mini Cheetah.
It includes custom made power electronics with messages transmitting informations at 1Khz.

For each actuator
Command:
 * position
 * velocity
 * current (current or equivalent ) as a feedforward term
 * current limit (which can evolve according to the trajectory)
 * Kp
 * Kd
Sensor:
 * position
 * velocity
 * current
 * contact
 * resistance (to evaluate temperature)
 * contact (with the ground)

IMU:
 * angular velocity
 * acceleration
 * acceleration without gravity
 * orientation (provided by an embedded EKF)

More details at https://open-dynamic-robot-initiative.github.io/

### CODE REVIEW

File: design_drafts/components_architecture_and_urdf_examples.md [hunk 1]
```diff
@@ -144,6 +144,287 @@ Note:
...
+      <param name="example_param_read_for_sec">0.001</param>
+    </hardware>
+    <joint name="front_left_leg_joint1">
+      <classType>ros2_control_components/MultiInterfaceJoint_MultiWrite</classType>
+      <commandInterfaceType>position</commandInterfaceType>
+      <commandInterfaceType>velocity</commandInterfaceType>
+      <commandInterfaceType>effort</commandInterfaceType>
+      <commandInterfaceType>Kp</commandInterfaceType>
+      <commandInterfaceType>Kd</commandInterfaceType>
+      <commandInterfaceType>effort_limit</commandInterfaceType>
```
Comment (bmagyar) [inline design_drafts/components_architecture_and_urdf_examples.md:161]: please follow the guidance on this ticket
https://github.com/ros-controls/roadmap/issues/23

Comment (destogl) [reply design_drafts/components_architecture_and_urdf_examples.md:161]: You can check #25 for examples

File: design_drafts/components_architecture_and_urdf_examples.md [hunk 2]
```diff
@@ -144,6 +144,287 @@ Note:
   </ros2_control>
 ```
 
+```xml
+  <ros2_control name="12DOF_Legged_System_Robot_MultiInterface" type="system">
```
Comment (destogl) [inline design_drafts/components_architecture_and_urdf_examples.md:148]: Please use capitalized CamelCase in the names. We will also change this in #25 

e.g. "12DoFLeggedSystemRobotMultiInterface"

File: design_drafts/components_architecture_and_urdf_examples.md [hunk 3]
```diff
@@ -144,6 +144,287 @@ Note:
...
+    <hardware>
+      <classType>ros2_control_demo_hardware/2DOF_System_Hardware_MultiInterface_MultiWrite</classType>
```
Comment (destogl) [inline design_drafts/components_architecture_and_urdf_examples.md:150]: here also capitalized CamelCase:

"2DoFSystemHardwareMultiInterfaceMultiWrite"

Comment (destogl) [reply design_drafts/components_architecture_and_urdf_examples.md:150]: Oh, here we should use different name:

either: "ros2_control_demo_hardware/12DoFSystemHardwareMultiInterfaceMultiWrite"

or event better:

"talos_control/TalosSystemHardware" or "talos_control/TalosRobotHardware" (choose one you prefer)

Of course we should than create `talos_control` package at some point.

File: design_drafts/components_architecture_and_urdf_examples.md [hunk 4]
```diff
@@ -144,6 +144,287 @@ Note:
...
+      <param name="example_param_write_for_sec">0.001</param>
+      <param name="example_param_read_for_sec">0.001</param>
+    </hardware>
+    <joint name="front_left_leg_joint1">
+      <classType>ros2_control_components/MultiInterfaceJoint_MultiWrite</classType>
```
Comment (destogl) [inline design_drafts/components_architecture_and_urdf_examples.md:155]: also capitalized CamelCase:

"MultiInterfaceMultiWriteJoint" (we will rename this in #25)

File: design_drafts/components_architecture_and_urdf_examples.md [hunk 5]
```diff
@@ -144,6 +144,287 @@ Note:
...
+      <commandInterfaceType>position</commandInterfaceType>
+      <commandInterfaceType>velocity</commandInterfaceType>
+      <commandInterfaceType>effort</commandInterfaceType>
+      <commandInterfaceType>Kp</commandInterfaceType>
```
Comment (destogl) [inline design_drafts/components_architecture_and_urdf_examples.md:159]: also add here some min/max values is such exist.

**Review by destogl — CHANGES_REQUESTED** [review]
Some changes w.r.t #25

File: design_drafts/components_architecture_and_urdf_examples.md [hunk 6]
```diff
@@ -108,7 +108,7 @@ Note:
   * the communication is done using proprietary API to communicate with robot control box
   * Data for all joints is exchanged in batch (at once)
   * Multiple values can be commanded, e.g. goal position and the maximal velocity on the trajectory allowed
-  * Examples: (humanoid robots?)
+  * Examples: humanoid robots (TALOS), 4 legged robot [solo](https://github.com/open-dynamic-robot-initiative/master-board/blob/master/documentation/BLMC_%C2%B5Driver_SPI_interface.md)
```
[the remark this answers is not in the document]
Comment (olivier-stasse) [reply design_drafts/components_architecture_and_urdf_examples.md:111]: Good to me. To be a bit more precise, TALOS and the 4 legged robot (SOLO) are two different robots. SOLO is mostly to highlight an original way to access the low level power electronics.

File: design_drafts/components_architecture_and_urdf_examples.md (hunk 3 above)
Comment (olivier-stasse) [reply design_drafts/components_architecture_and_urdf_examples.md:150]: Will do

**Review by bmagyar — APPROVED** [review]

**Comment by destogl** [comment]
@bmagyar  Thanks for merging this!

**Pull request merged**